### PR TITLE
fix: Add content-type support

### PR DESCRIPTION
**Pull request by thomasheartman**
**tl;dr:**
- Add content-type middleware
- Reject (415) requests with wrong content-type
- Assume that if there is no content-type, it should be JSON

These changes enable content-type validation for the proxy by using a piece of [express middleware from Unleash](https://github.com/Unleash/unleash/blob/dab55bf3c6ecc66049b5e645ef968fa76991f22f/src/lib/middleware/content_type_checker.ts). The goal is to ensure that requests made with the wrong content-type are told that we don't accept that content-type (415 response). 

It also adds 415 responses to the OpenAPI description for the relevant endpoints.

## Changes from Unleash's content-type checker

The content-type checker has been modified slightly from Unleash's content-type checker:
1. If the request has no body, the middleware is a no-op. This means we can apply the middleware to all operations, and it's not going to add content-types to GET requests or DELETE requests without bodies, for instance. It's also not going to add anything to POST requests without bodies.
2. If the request has a body, but no content-type, the middleware assumes you forgot to add it (**be liberal in what you accept**) and will add the first accepted content-type. This might not work in all cases, but should cover most use cases where that happens.

As with the other version of the middleware, requests with wrong content-types are rejected.

## Files

Most of the changes should be small enough to inspect easily, but there's two files that are new. They have been adapted and taken from Unleash. Here's a description of what I've done to them:

### `src/content-type-checker.ts`

This is adapted from the [Unleash-version of the middleware](https://github.com/Unleash/unleash/blob/dab55bf3c6ecc66049b5e645ef968fa76991f22f/src/lib/middleware/content_type_checker.ts). Some slight changes to the logic and to the description.

### `src/test/content-type-checker.test.ts`

Like with the content-type-checker itself, this is largely just copied from Unleash, but there's some new tests that verify that adding the default content-type works as expected. They are in the `describe('add default content-types to requests with bodies but no content-type'` block. 

There are also some changes in the mock request creation logic to facilitate mocking with and without request bodies.

## Discussion point

Where in the chain does the middleware go? I've added it semi-randomly. Should it go somewhere else in particular or is it not that important?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,19 +44,21 @@
     "express": "^4.18.2",
```
[the remark this answers is not in the document]
Comment (thomasheartman) [reply package.json:44]: Thanks, liftbot! 

I'm sure this is useful, but I don't think it's relevant to this PR. Reviewer thoughts? Can we ignore for now?

[the remark this answers is not in the document]
Comment (gastonfournier) [reply package.json:44]: Agree! @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (thomasheartman) [reply package.json:44]: @sonatype-lift ignore

File: src/test/unleash-proxy.test.ts [hunk 2]
```diff
@@ -684,3 +684,65 @@ test('Should return 501 when all feature toggles is not enabled', () => {
...
+                    enableAllEndpoint: true,
+                },
+                client,
+            );
+            client.emit('ready');
+
+            await request(app)
+                .post(endpoint)
+                .set('Authorization', 'sdf')
+                .set('Content-Type', 'application/json')
```
Comment (gastonfournier) [inline src/test/unleash-proxy.test.ts:712]: Is this well aligned with the test description? 
> Should assign default content-type if the request has a body but no content-type (%s)

According to that, I think we should not have this line

**Review by gastonfournier — APPROVED** [review]
Just a minor question, but looks good!

File: src/test/unleash-proxy.test.ts (hunk 2 above)
Comment (thomasheartman) [reply src/test/unleash-proxy.test.ts]: Aha! You're perfectly right! That's a holdover from when I couldn't make it work. It's been addressed in 849084ebf735f74b2379007fe52aa9767f9f0ff2
